File: VB/E5030/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports E5030.Reports

Namespace E5030.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View()
		End Function

		<HttpGet> _
		Public Function ReportCallbackPanelPartial() As ActionResult
			Dim report As New ProductsByCategoriesReport()
			Return PartialView("_ReportCallbackPanelPartial", report)
		End Function

		<HttpPost> _
		Public Function ReportCallbackPanelPartial(ByVal categoryID As Integer) As ActionResult
			Dim report As New ProductsByCategoriesReport()
			report.CategoryExpandClicked(categoryID)
			Return PartialView("_ReportCallbackPanelPartial", report)
		End Function

		Public Function DocumentViewerPartial() As ActionResult
			Dim report As New ProductsByCategoriesReport()
			Return PartialView("_DocumentViewerPartial", report)
		End Function

		Public Function DocumentViewerExport() As ActionResult
			Dim report As New ProductsByCategoriesReport()
			Return ReportViewerExtension.ExportTo(report)
		End Function
	End Class
End Namespace

File: CS/E5030/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using E5030.Reports;

namespace E5030.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        [HttpGet]
        public ActionResult ReportCallbackPanelPartial() {
            ProductsByCategoriesReport report = new ProductsByCategoriesReport();
            return PartialView("_ReportCallbackPanelPartial", report);
        }

        [HttpPost]
        public ActionResult ReportCallbackPanelPartial(int categoryID) {
            ProductsByCategoriesReport report = new ProductsByCategoriesReport();
            report.CategoryExpandClicked(categoryID);
            return PartialView("_ReportCallbackPanelPartial", report);
        }

        public ActionResult DocumentViewerPartial() {
            ProductsByCategoriesReport report = new ProductsByCategoriesReport();
            return PartialView("_DocumentViewerPartial", report);
        }

        public ActionResult DocumentViewerExport() {
            ProductsByCategoriesReport report = new ProductsByCategoriesReport();
            return ReportViewerExtension.ExportTo(report);
        }        
    }
}